lib/utils/tty.js

```diff
@@ -5,9 +5,27 @@
 var _ = require("lodash");
 var readline = require("readline");
 var cliWidth = require("cli-width");
+var utils = require("./utils");
 
 var tty = module.exports;
 
+/**
+ * Get the number of lines the current line occupies
+ * @return {Number}  The number of lines
+ */
+
+tty.lines = function() {
+  var value = this.rl.line || this.rl.history[0];
+  var lineLength = utils.getUnicodeStringLength( this.rl._prompt + value );
+  var result = lineLength / cliWidth();
```

Comment:
@SBoudrias I've tested in Windows 8.1 and `cliWidth` returns the correct width on that platform. I had to implement some windows-specific workaround for me patch to work fine, but I can confirm it works as expected on `cmd.exe`.

There still exist the possibility of `cliWidth()` returning 0, maybe in older Windows versions? I propose checking the return value of `cliWidth`, and if it is zero, don't make use of my patch, thus resulting in the actual behaviour. What do you think?